src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -125,8 +125,53 @@ class spell_monk_crackling_jade_lightning_knockback_proc_aura : public SpellScri
     }
 };
 
+// Provoke - 115546
+class spell_monk_provoke : public SpellScriptLoader
+{
+    public:
+    spell_monk_provoke() : SpellScriptLoader("spell_monk_provoke")
+    { }
+
+    class spell_monk_provoke_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_monk_provoke_SpellScript);
+
+        SpellCastResult CheckCast()
+        {
+            Unit* target = GetExplTargetUnit();
+            if (!target)
+                return SPELL_FAILED_NO_VALID_TARGETS;
+            else if (target->GetTypeId() == TYPEID_PLAYER)
+                return SPELL_FAILED_BAD_TARGETS;
+            else if (!target->IsWithinLOSInMap(GetCaster()))
+                return SPELL_FAILED_LINE_OF_SIGHT;
+            return SPELL_CAST_OK;
+        }
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetCaster())
+                if (caster->getClass() == CLASS_MONK && caster->GetTypeId() == TYPEID_PLAYER)
```

Comment:
Oh and:
It's a dummy effect because if you are brewmaster and you have a statue up, the statue will cast the AOE version of Provoke.

> This ability can be targeted on your Statue of the Black Ox, causing the same effect on all enemies within  10 yards of the statue